src/exchanges/cache.ts

```diff
@@ -145,19 +146,23 @@ const afterQuery = (
   resultCache: ResultCache,
   operationCache: OperationCache
 ) => (response: OperationResult) => {
-  const {
-    operation: { key },
-    data,
-  } = response;
+  const { operation, data } = response;
+
   if (data === undefined) {
     return;
   }
 
-  resultCache.set(key, response);
+  resultCache.set(operation.key, {
```

Comment:
Let's modify the operation only when it goes back out, so that we don't get this confused later on: https://github.com/FormidableLabs/urql/blob/113b732bd1a3dd32e5b1a9df542a7a641e82acde/src/exchanges/cache.ts#L65

Which would also allow us to mark this line as a cache miss: https://github.com/FormidableLabs/urql/blob/113b732bd1a3dd32e5b1a9df542a7a641e82acde/src/exchanges/cache.ts#L69